hack/hive-generate-config.sh

```diff
@@ -1,11 +1,13 @@
 #!/bin/bash
 
+HIVE_REPO="${HIVE_REPO:=https://github.com/openshift/hive.git}"
+
 # This is the commit sha that the image was built from and ensures we use the correct configs for the release
-HIVE_IMAGE_COMMIT_HASH="c63c9b0"
+HIVE_IMAGE_COMMIT_HASH="${HIVE_IMAGE_COMMIT_HASH:=c63c9b0}"
 
 # For now we'll use the quay hive image, but this will change to an ACR once the quay.io -> ACR mirroring is setup
 # Note: semi-scientific way to get the latest image: `podman search --list-tags --limit 10000 quay.io/app-sre/hive | tail -n1`
-HIVE_IMAGE="quay.io/app-sre/hive:${HIVE_IMAGE_COMMIT_HASH}"
+HIVE_IMAGE="${HIVE_IMAGE:=quay.io/app-sre/hive:${HIVE_IMAGE_COMMIT_HASH}}"
```

Comment:
The way this variable and `$HIVE_IMAGE_COMMIT_HASH` are interacting is a tad confusing. If this gets set to `$IMG` like in the instructions then `$HIVE_IMAGE_COMMIT_HASH` isn't used at all for the image tag since `$IMG` has `latest`. If this is unset but `$HIVE_IMAGE_COMMIT_HASH` is, then the hash being used is likely one from a local git branch and the tag won't exist in the `quay.io/app-sre/hive` container repository. Other functions in this file use `$HIVE_IMAGE_COMMIT_HASH` to generate configs, and having the image tag default to `latest` could cause some problems when the `hive:latest` and `hive:${HIVE_IMAGE_COMMIT_HASH}` are not the same image.

Reply:
Yes, it is confusing. Any suggestions?


Keep in mind - this is for rare cases when you need to test a custom image. In theory `HIVE_IMAGE_COMMIT_HASH` only needs to be updated when CRDs/manifests change. In most of the cases it will be ok to use `HIVE_IMAGE_COMMIT_HASH` for config generation and using `HIVE_IMAGE` from a different commit.

Reply:
> In theory HIVE_IMAGE_COMMIT_HASH only needs to be updated when CRDs/manifests change. In most of the cases it will be ok to use HIVE_IMAGE_COMMIT_HASH for config generation and using HIVE_IMAGE from a different commit.

I think leaving this as a note in the docs and/or as a comment in the script would be the best approach then.